user_sync/connector/umapi.py

```diff
@@ -146,6 +146,32 @@ def iter_users(self):
         except umapi_client.UnavailableError as e:
             raise AssertionException("Error contacting UMAPI server: %s" % e)
 
+    def get_groups(self):
+        return list(self.iter_groups())
+
+    def iter_groups(self):
+        try:
+            for g in umapi_client.GroupsQuery(self.connection):
+                yield g
+        except umapi_client.UnavailableError as e:
+            raise AssertionException("Error contacting UMAPI server: %s" % e)
+
+    def get_user_groups(self):
+        return list(self.iter_user_groups())
+
+    def iter_user_groups(self):
+        try:
+            for g in umapi_client.UserGroupsQuery(self.connection):
+                yield g
+        except umapi_client.UnavailableError as e:
+            raise AssertionException("Error contacting UMAPI server: %s" % e)
+
+    def create_group(self, name):
```

Comment:
I don't think it would make sense to do that here.  Auto-creation is designed to be a separate feature from the additional group discovery/mapping feature.  It can be enabled independently of it and be used solely to auto-create Adobe groups targeted in the group mapping and/or the extension config.  (conversely, the "additional groups" option can be enabled without group auto-creation)

It would probably make more sense to log the "additional group" mapping somewhere in the additional group resolution workflow.

Reply:
So then make the source group name be an optional argument and don't use it if it's not passed.  The comment you're currently using is fairly useless in the auto-mapping case and would be a lot better if it had the source group.